tools/msgdis.py

```diff
@@ -0,0 +1,442 @@
+#!/usr/bin/env python3
+# 
+# message_data_static disassembler/decompiler
+# 
+
+import re, struct
+from os import path
+
+# ===================================================
+#   Util
+# ===================================================
+
+def as_hword_list(b):
+    if len(b) % 2 != 0:
+        return None
+    return [h[0] for h in struct.iter_unpack(">H", b)]
+
+def as_word_list(b):
+    if len(b) % 4 != 0:
+        return None
+    return [i[0] for i in struct.iter_unpack(">I", b)]
+
+def as_message_table_entry(b):
+    if len(b) % 8 != 0:
+        return None
+    return [(e[0], e[1]>>0x4&0xF, e[1]&0xF, e[2]) for e in [i for i in struct.iter_unpack(">HBxI", b)]]
+
+def segmented_to_physical(address):
+    return address & ~0x07000000
+
+# ===================================================
+#   Decode message_data_static encoded strings
+# ===================================================
+
+"""
+Special characters conversion
+"""
+charmap = {
+    0x7F: '‾',
+    0x80: 'À',
+    0x81: 'î',
+    0x82: 'Â',
```

Comment:
wouldn't it be better to not have this duplicated with `assets/text/charmap.txt`
also about `assets/text/charmap.txt` I think a more common format like json should be used if you keep that file

Reply:
This is the charmap used by the base game, but you could imagine a situation where a user might want to change the encoding for whatever reason. If that were to happen it would break extraction, so imo they should be duplicated and the one in `msgdis` shouldn't be touched by modifications while the charmap.txt should.

As for json, I don't see how it's at all beneficial, sounds like it's just overcomplicating it for no reason.

Reply:
json doesn't support hexadecimal numbers, and when I create a json programmatically from the extraction charmap it replaces several of the characters with their escaped unicode form "\u...." so I think I'll stick with how it is currently.